### PR TITLE
fix: dashboard promote buttons disabled during deploy

### PR DESCRIPTION
**Pull request by bpoland**
- keep promote and abort buttons enabled while deploying
- disable them at all other times
- also, only enable retry button when Rollout is degraded

Fixes https://github.com/argoproj/argo-rollouts/issues/1581

Signed-off-by: Ben Poland <ben.poland@faire.com>

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-rollouts/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [ ] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [ ] I've signed my commits with [DCO](https://github.com/argoproj/argoproj)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My builds are green. Try syncing with master if they are not. 
* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo-rollouts/blob/master/USERS.md).

### CODE REVIEW

File: ui/src/app/components/rollout-actions/rollout-actions.tsx [hunk 1]
```diff
@@ -36,6 +37,7 @@ export const RolloutActionButton = (props: {action: RolloutAction; rollout: Roll
                 label: 'RETRY',
                 icon: 'fa-redo-alt',
                 action: api.rolloutServiceRetryRollout,
+                disabled: props.rollout.status !== RolloutStatus.Degraded,
```
Comment (harikrongali) [inline ui/src/app/components/rollout-actions/rollout-actions.tsx:40]: why is this added? you can retry degraded rollout as well

Comment (bpoland) [reply ui/src/app/components/rollout-actions/rollout-actions.tsx:40]: Yes I think the button should only be enabled when the rollout is degraded, right?  The double negative is a little confusing :)

Comment (harikrongali) [reply ui/src/app/components/rollout-actions/rollout-actions.tsx:40]: i misread the line..yes you are correct

**Review by harikrongali — APPROVED** [review]

**Review by alexmt — APPROVED** [review]
